// react-native-list-view-select/ListPopover.js
import React, { Component } from 'react';
import { StyleSheet, Dimensions, View, Text, TouchableHighlight } from 'react-native';
import I18n from '../../libs/I18n';
import Icon from 'react-native-vector-icons/FontAwesome';
import ListPopover from './ListPopover';
import _ from 'lodash';

export default class Heading extends Component {

  changeCity() {
    alert('uo');
  }

  constructor(props) {
    super(props);
    this.state = {
      item: "Select Item",
      isVisible: false,
    };
    _.bindAll(this, ['showPopover', 'closePopover', 'setItem']);
  }

  showPopover() {
    this.setState({isVisible: true});
  }

  closePopover() {
    this.setState({isVisible: false});
  }

  setItem(item) {
    this.setState({item: item});
  }


  render() {
    const { selectedCity } = this.props;
    I18n.nameSpace('register');
    var items = [
      "Item 1",
      "Item 2",
      "Item 3",
      "Item 4",
    ];

    return (
      <View style={styles.container}>
        <TouchableHighlight style={styles.button} onPress={this.showPopover}>
          <Text>{this.state.item}</Text>
        </TouchableHighlight>
        <ListPopover
          list={items}
          isVisible={this.state.isVisible}
          onClick={this.setItem}
          onClose={this.closePopover}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    paddingTop: 100,
    paddingBottom: 100,
  },
  text: {
    color: '#FFF',
    fontSize: 15,
    fontFamily: 'Source Sans Pro',
    lineHeight: 15
  },
  selectedCityText: {
    color: '#75c58c',
    fontSize: 20,
    paddingRight: 10,
  },
  selectedCity: {
    flexDirection: 'row',
  },
});
